misc/python/materialize/data_ingest/definition.py

```diff
@@ -0,0 +1,185 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from copy import deepcopy
+from enum import Enum
+from typing import List, Optional
+
+from materialize.data_ingest.data_type import RecordSize
+from materialize.data_ingest.field import Field
+from materialize.data_ingest.row import Operation, Row
+from materialize.data_ingest.rowlist import RowList
+from materialize.data_ingest.transaction import Transaction
+
+
+class Records(Enum):
+    ONE = 1
+    MANY = 2
+    ALL = 3
+
+
+class Keyspace(Enum):
+    SINGLE_VALUE = 1
+    LARGE = 2
+    EXISTING = 3
+
+
+class Target(Enum):
+    KAFKA = 1
+    POSTGRES = 2
+    PRINT = 3
+
+
+class Definition:
+    def generate(self, fields: List[Field]) -> List[Transaction]:
+        raise NotImplementedError
+
+
+class Insert(Definition):
+    def __init__(self, count: Records, record_size: RecordSize):
+        self.count = count
+        self.record_size = record_size
+        self.current_key = 0
+
+    def max_key(self) -> int:
+        if self.count == Records.ONE:
+            return 1
+        elif self.count == Records.MANY:
+            return 1000
+        else:
+            raise ValueError(f"Unexpected count {self.count}")
+
+    def generate(self, fields: List[Field]) -> List[Transaction]:
+        if self.count == Records.ONE:
+            count = 1
+        elif self.count == Records.MANY:
+            count = 1000
+        else:
+            raise ValueError(f"Unexpected count {self.count}")
+
+        fields_with_values = deepcopy(fields)
+
+        transactions = []
+        for i in range(count):
+            for field in fields_with_values:
+                if field.is_key:
+                    field.value = field.typ.num_value(self.current_key)
+                else:
+                    field.value = field.typ.random_value(self.record_size)
+            self.current_key += 1
```

Comment:
Random keys will be nice at some future stage. UUIDs for example. Otherwise RocksDB and other pieces of code will have the luxury of lining things up in nice order in various data structures.

For example, you can keep the concept of `current_key` but use it as a seed to generate a UUID, so that you get predictability and randomness at the same time.

Reply:
I'll add the preparation for that.